src/server/qgsserverogcapi.h

```diff
@@ -0,0 +1,203 @@
+
+    /**
+     * Returns a map of contentType => mime type
+     * \note not available in Python bindings
+     */
+    static const QMap<QgsServerOgcApi::ContentType, QString> contentTypeMimes() SIP_SKIP;
+
+    /**
+     * Returns contenType specializations (e.g. JSON => [GEOJSON, OPENAPI3], XML => [GML])
+     * \note not available in Python bindings
+     */
+    static const QHash<QgsServerOgcApi::ContentType, QList<QgsServerOgcApi::ContentType> > contentTypeAliases() SIP_SKIP;
+
+    // Utilities
+#ifndef SIP_RUN
+
+    /**
+     * Register an OGC API handler passing \a Args to the constructor
+     * \note not available in Python bindings
+     */
+    template<class T, typename... Args>
+    void registerHandler( Args... args )
+    {
+      mHandlers.emplace_back( std::make_shared<T>( args... ) );
+    }
+#endif
+
+    /**
+     * Registers an OGC API \a handler, ownership of the handler is transferred to the API
+     */
+    void registerHandler( QgsServerOgcApiHandler *handler SIP_TRANSFER );
+
+    /**
+     * Returns a sanitized \a url with extra slashes removed
+     */
+    static QUrl sanitizeUrl( const QUrl &url );
+
+    /**
+     * Returns the string representation of \a rel attribute.
+     */
+    static std::string relToString( const QgsServerOgcApi::Rel &rel );
+
+    /**
+     * Returns the string representation of a \a ct (Content-Type) attribute.
+     */
+    static QString contentTypeToString( const QgsServerOgcApi::ContentType &ct );
+
+    /**
+     * Returns the string representation of a \a ct (Content-Type) attribute.
+     */
+    static std::string contentTypeToStdString( const QgsServerOgcApi::ContentType &ct );
+
+    /**
+     * Returns the file extension for a \a ct (Content-Type).
+     */
+    static QString contentTypeToExtension( const QgsServerOgcApi::ContentType &ct );
+
+    /**
+     * Returns the Content-Type value corresponding to \a extension.
+     */
+    static QgsServerOgcApi::ContentType contenTypeFromExtension( const std::string &extension );
+
+    /**
+     * Returns the mime-type for the \a contentType or an empty string if not found
+     */
+    static std::string mimeType( const QgsServerOgcApi::ContentType &contentType );
+
+    /**
+     * Returns the absolute path to base directory where resources for
+     * OGC APIs are stored.
+     *
+     * TODO: make this path configurable by env and/or settings
+     */
+    static const QString resourcesPath();
+
+    /**
+     * Returns registered handlers
+     */
+    const std::vector<std::shared_ptr<QgsServerOgcApiHandler> > handlers() const SIP_SKIP;
+
+  private:
+
+    QString mRootPath;
+    QString mName;
+    QString mDescription;
+    QString mVersion;
+
+    //Note: this cannot be unique because of SIP bindings
+    std::vector<std::shared_ptr<QgsServerOgcApiHandler>> mHandlers;
+
+    //! Stores content type mime strings
+    static QMap<QgsServerOgcApi::ContentType, QString> sContentTypeMime;
```

Comment:
Is it really necessary to have `static` members?

Reply:
Of course not, but they are not going to change during the life of the application, I think it's more efficient to have them build as static. Do you see any potential issue here?